OpenAPI/kiota/tutorials/dotnet-dependency-injection.md

```diff
@@ -0,0 +1,301 @@
+---
+title: Use Kiota with dependency injection in .NET
+description: Learn how you can combine Kiota with dependency injection in .NET to build API clients.
+author: svrooij
+ms.author: vibiret
+ms.topic: tutorial
+date: 05/10/2024
+---
+
+# Register a Kiota API client in .NET with dependency injection
+
+In this tutorial, you will learn how to register a [Kiota API client](/openapi/kiota/tutorials/dotnet-azure?wt.mc_id=SEC-MVP-5004985) with [dependency injection](/dotnet/core/extensions/dependency-injection-usage?wt.mc_id=SEC-MVP-5004985).
+
+## Required tools
+
+- [.NET SDK 8.0](https://get.dot.net/8)
+- [Kiota CLI](/openapi/kiota/install?tabs=bash&wt.mc_id=SEC-MVP-5004985#install-as-net-tool)
+
+## Create a project
+
+Run the following command in the directory you want to create a new project.
+
+```dotnetcli
+dotnet new webapi -o kiota-with-di
+cd kiota-with-di
+dotnet new gitignore
+```
+
+## Add dependencies
+
+Before you can compile and run the generated API client, ensure the generated source files are part of a project with the required dependencies. Your project must reference the [abstraction package](https://github.com/microsoft/kiota-abstractions-dotnet) and default implementations.
+
+- HTTP ([Kiota default HttpClient-based implementation](https://github.com/microsoft/kiota-http-dotnet)) version `1.4.2` or higher
+- JSON serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-json-dotnet))
+- Form serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-form-dotnet))
+- Text serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-text-dotnet))
+- Multipart serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-multipart-dotnet))
+- HttpClient support for Dependency Injection [Microsoft.Extensions.Http](https://www.nuget.org/packages/Microsoft.Extensions.Http)
+
+Run the following commands to get the required dependencies.
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.Http
+dotnet add package Microsoft.Kiota.Abstractions
+dotnet add package Microsoft.Kiota.Http.HttpClientLibrary
+dotnet add package Microsoft.Kiota.Serialization.Json
+dotnet add package Microsoft.Kiota.Serialization.Form
+dotnet add package Microsoft.Kiota.Serialization.Text
+dotnet add package Microsoft.Kiota.Serialization.Multipart
+```
+
+## Generate the API client
+
+Kiota generates API clients from OpenAPI documents. Create a file named _github-releases.yml_ and add the following.
```

Comment:
have we considered using the real description from the search results instead and using filters?

Reply:
What do you propose?